pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -53,74 +53,71 @@ const (
 // which is what is expected when interacting with libcontainer
 var hugePageSizeList = []string{"B", "kB", "MB", "GB", "TB", "PB"}
 
-// ConvertCgroupNameToSystemd converts the internal cgroup name to a systemd name.
-// For example, the name /Burstable/pod_123-456 becomes Burstable-pod_123_456.slice
-// If outputToCgroupFs is true, it expands the systemd name into the cgroupfs form.
-// For example, it will return /Burstable.slice/Burstable-pod_123_456.slice in above scenario.
-func ConvertCgroupNameToSystemd(cgroupName CgroupName, outputToCgroupFs bool) string {
-	name := string(cgroupName)
-	result := ""
-	if name != "" && name != "/" {
-		parts := strings.Split(name, "/")
-		results := []string{}
-		for _, part := range parts {
-			// ignore leading stuff
-			if part == "" {
-				continue
-			}
-			// detect if we are given a systemd style name.
-			// if so, we do not want to do double encoding.
-			if IsSystemdStyleName(part) {
-				part = strings.TrimSuffix(part, systemdSuffix)
-				separatorIndex := strings.LastIndex(part, "-")
-				if separatorIndex >= 0 && separatorIndex < len(part) {
-					part = part[separatorIndex+1:]
-				}
-			} else {
-				// systemd treats - as a step in the hierarchy, we convert all - to _
-				part = strings.Replace(part, "-", "_", -1)
-			}
-			results = append(results, part)
+var RootCgroupName = CgroupName([]string{})
+
+// NewCgroupName composes a new cgroup name.
+// Use RootCgroupName as base to start at the root.
+// This function does some basic check for invalid characters at the name.
+func NewCgroupName(base CgroupName, components ...string) CgroupName {
+	for _, component := range components {
+		// Forbit using "_" in internal names. When remapping internal
+		// names to systemd cgroup driver, we want to remap "-" => "_",
+		// so we forbid "_" so that we can always revert the mapping.
+		if strings.Contains(component, "/") || strings.Contains(component, "_") {
+			panic(fmt.Errorf("Invalid character in component [%s] of CgroupName", component))
```

Comment:
nits: 
Error start with lower case: `s/Invalid/invalid`
Use `%q` to print the string: `s/%s/%q`

Reply:
Done.